test/simple/test-tls-cipher-list.js

```diff
@@ -44,6 +44,21 @@ function doTest(checklist, env, useswitch) {
       // Test NODE_LEGACY_CIPHER_LIST
       if (env) options = {env:{"NODE_LEGACY_CIPHER_LIST": env}};
       break;
+    case 4:
+      // Test NODE_CIPHER_LIST takes precedence over --cipher-list
+      args.unshift('--cipher-list=XYZ');
+      if (env) options = {env:{"NODE_CIPHER_LIST": env}};
+      break;
+    case 5:
+      // Test --enable-legacy-cipher-list takes precedence over --cipher-list
+      args.unshift('--cipher-list=XYZ');
+      args.unshift('--enable-legacy-cipher-list=' + env);
+      break;
+    case 6:
+      // Test NODE_LEGACY_CIPHER_LIST takes precendence over --cipher-list
+      args.unshift('--cipher-list=XYZ');
+      if (env) options = {env:{"NODE_LEGACY_CIPHER_LIST": env}};
+      break;
```

Comment:
We would probably need to have more tests than that to test the whole matrix of command line switches and environment variables. I would do that in a separate PR though, as this one is about `ls: throw an error on getLegacyCipher`.